spec/views/hyrax/base/_form.html.erb_spec.rb

```diff
@@ -1,80 +1,127 @@
-    # TODO: stub_model is not stubbing new_record? correctly on ActiveFedora models.
-    allow(work).to receive(:new_record?).and_return(true)
-    allow(work).to receive(:member_ids).and_return([1, 2])
-    allow(view).to receive(:curation_concern).and_return(work)
     allow(controller).to receive(:current_user).and_return(stub_model(User))
     assign(:form, form)
-    allow(controller).to receive(:controller_name).and_return('batch_uploads')
-    allow(controller).to receive(:action_name).and_return('new')
-    allow(controller).to receive(:repository).and_return(Hyrax::GenericWorksController.new.repository)
-    allow(controller).to receive(:blacklight_config).and_return(Hyrax::GenericWorksController.new.blacklight_config)
-
-    allow(form).to receive(:permissions).and_return([])
-    allow(form).to receive(:visibility).and_return('public')
-    stub_template 'hyrax/base/_form_files.html.erb' => 'files'
+    allow(controller).to receive(:action_name).and_return(controller_action)
+    allow(controller).to receive(:repository).and_return(controller_class.new.repository)
+    allow(controller).to receive(:blacklight_config).and_return(controller_class.new.blacklight_config)
   end
 
-  context "for a new object" do
-    let(:work) { GenericWork.new }
+  context 'with a change_set style form' do
+    let(:form) { Hyrax::Forms::ResourceForm.for(work) }
+    let(:work) { build(:monograph, title: 'comet in moominland') }
 
-    context 'with batch_upload on' do
-      before do
-        allow(Flipflop).to receive(:batch_upload?).and_return(true)
+    before do
+      # TODO: unstub these when they are supported by the form
+      stub_template('hyrax/base/_form_member_of_collections.html.erb' => 'collection membership')
+    end
+
+    context 'for a new object' do
+      it 'renders a form' do
         render
+
+        expect(rendered).to have_selector("form[action='/concern/monographs']")
       end
-      it 'shows batch uploads' do
-        expect(rendered).to have_link('Batch upload', href: hyrax.new_batch_upload_path(payload_concern: 'GenericWork'))
-        expect(rendered).to have_selector("form[action='/concern/generic_works'][data-param-key='generic_work']")
-        # Draws the "Share" tab, with data for the javascript.
-        expect(rendered).to have_selector('#share[data-param-key="generic_work"]')
+
+      context 'with batch_upload off' do
+        before do
+          allow(Flipflop).to receive(:batch_upload?).and_return(false)
+        end
+
+        it 'hides batch uploads' do
+          render
+          expect(rendered).not_to have_link('Batch upload', href: hyrax.new_batch_upload_path(payload_concern: 'GenericWork'))
+        end
       end
     end
 
-    context 'with batch_upload off' do
-      before do
-        allow(Flipflop).to receive(:batch_upload?).and_return(false)
+    context 'with an existing object' do
+      let(:work) { FactoryBot.valkyrie_create(:monograph) }
+
+      xit 'renders a form' do
```

Comment:
Is this expected? Maybe a comment as to why it's a `xit`

Reply:
there's some discussion on this in the commit message. passing this test requires grappling with `#version`/optimistic locking, which i'm considering out of scope here.